e2etest/zt_newe2e_list_test.go

```diff
@@ -45,9 +53,261 @@ func (s *ListSuite) Scenario_ListBasic(svm *ScenarioVariationManager) {
 			},
 		})
 
-	expectedObjects := map[AzCopyOutputKey]cmd.AzCopyListObject{
-		AzCopyOutputKey{Path: "test"}: {Path: "test", ContentLength: "1.00 KiB"},
+	ValidateListOutput(svm, stdout, expectedObjects, nil)
+}
+
+func (s *ListSuite) Scenario_ListHierarchy(svm *ScenarioVariationManager) {
+	acctService := ResolveVariation(svm, []struct {
+		acct string
+		loc  common.Location
+	}{
+		{PrimaryStandardAcct, common.ELocation.Blob()},
+		{PrimaryStandardAcct, common.ELocation.File()},
+		{PrimaryHNSAcct, common.ELocation.BlobFS()},
+	})
+
+	acct := GetAccount(svm, acctService.acct)
+	srcService := acct.GetService(svm, acctService.loc)
+
+	svm.InsertVariationSeparator(":")
+	srcContainer := CreateResource[ContainerResourceManager](svm, srcService, ResourceDefinitionContainer{})
+	var expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject
+	if srcService.Location() == common.ELocation.Blob() {
+		expectedObjects = map[AzCopyOutputKey]cmd.AzCopyListObject{}
+	} else {
+		expectedObjects = map[AzCopyOutputKey]cmd.AzCopyListObject{
+			AzCopyOutputKey{Path: "/"}: {Path: "/", ContentLength: "0.00 B"},
+		}
+	}
+	objects := []ResourceDefinitionObject{
+		{ObjectName: pointerTo("file_in_root.txt"), Body: NewRandomObjectContentContainer(svm, SizeFromString("1K")), Size: "1.00 KiB"},
+		{ObjectName: pointerTo("dir_in_root"), ObjectProperties: ObjectProperties{EntityType: common.EEntityType.Folder()}, Size: "0.00 B"},
+		{ObjectName: pointerTo("dir_in_root/file.txt"), Body: NewRandomObjectContentContainer(svm, SizeFromString("2K")), Size: "2.00 KiB"},
+		{ObjectName: pointerTo("dir_in_root/subdir"), ObjectProperties: ObjectProperties{EntityType: common.EEntityType.Folder()}, Size: "0.00 B"},
+	}
+	// Scale up from service to object
+	for _, o := range objects {
+		obj := CreateResource[ObjectResourceManager](svm, srcContainer, o)
+		name := obj.ObjectName()
+		if o.EntityType == common.EEntityType.Folder() {
+			name += "/"
+			if obj.Location() == common.ELocation.Blob() {
+				continue
+			}
+		}
+		expectedObjects[AzCopyOutputKey{Path: name}] = cmd.AzCopyListObject{Path: name, ContentLength: o.Size}
+	}
+
+	stdout, _ := RunAzCopy(
+		svm,
+		AzCopyCommand{
+			Verb: AzCopyVerbList,
+			Targets: []ResourceManager{
+				srcContainer.(RemoteResourceManager).WithSpecificAuthType(EExplicitCredentialType.SASToken(), svm, CreateAzCopyTargetOptions{
+					SASTokenOptions: GenericServiceSignatureValues{
+						ContainerName: srcContainer.ContainerName(),
+						Permissions:   (&blobsas.ContainerPermissions{Read: true, List: true}).String(),
+					},
+				}),
+			},
+			Flags: ListFlags{
+				GlobalFlags: GlobalFlags{
+					OutputType: to.Ptr(common.EOutputFormat.Json()),
```

Comment:
JSON should be default these days, I think? Check the flags file.

Reply:
yeah. I think there used to be a bug with the default output type. seems to be working fine now. I'll remove these.